File: Euler3/Euler3.java
public class Euler3 {
    private long maxPrimeFactor = 0;
    public static void main(String[] args){
        long num = 600851475143L;
        Euler3 e3 = new Euler3();
        e3.Factor(num);
        System.out.println(e3.maxPrimeFactor);
    }

    private void Factor(long num){
       for (int i = 2; i*i < num; i++){
            if (num % i == 0){
                System.out.println("factor:" + i);
                if(num/i > maxPrimeFactor)
                    Factor((long)(num/i));
                if(i > maxPrimeFactor)
                    Factor((long) i);
                return;
            }
       }
       //this means the Factor is prime, check if it's max
       maxPrimeFactor = num;
    }
    
}

File: Python/euler3.py
import math
#This program finds the biggest prime factor of the argument entered into euler3
factors = []
def euler3(n):
    numb = n
    for x in range(3, int(math.sqrt(numb))+1):
        if numb % x == 0:
            check = checkPrime(x)
            if check:
                factors.append(x)
                other = n//x
                if checkPrime(other):
                    factors.append(other)
                    break;
                euler3(other)
            else:
                euler3(x)
            
def checkPrime(n):
    for x in range(2, int(math.sqrt(n)) + 1):
        if n%x == 0:
            return False
    return True
def findBiggest():
    biggest = 0
    for x in factors:
        if x > biggest:
            biggest = x
    return biggest
euler3(600851475143)
print(findBiggest())